extensions/blocks/mailchimp/view.js

```diff
@@ -15,12 +15,16 @@ import './view.scss';
 
 const blockClassName = 'wp-block-jetpack-mailchimp';
 
-function fetchSubscription( blogId, email ) {
-	const url =
+function fetchSubscription( blogId, email, params ) {
+	let url =
 		'https://public-api.wordpress.com/rest/v1.1/sites/' +
 		encodeURIComponent( blogId ) +
 		'/email_follow/subscribe?email=' +
 		encodeURIComponent( email );
+
+	for ( const param in params ) {
+		url += '&' + param + '=' + params[ param ];
```

Comment:
Should these URL params be encoded with `encodeURIComponent`?

Reply:
Good call: https://github.com/Automattic/jetpack/pull/14050/commits/37539884b6b0cfaab8b9ce3a01b2aca1cccbe31a